lib/bundle/checker.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+module Bundle
+  module Checker
+    class Base
+      PACKAGE_TYPE = :pkg
+      PACKAGE_TYPE_NAME = "Package"
+
+      NO_ACTION = [].freeze
+
+      def action_required_for(formula)
```

Comment:
Sorry, I'm probably being stupid but I still don't understand how the action for a formula is the formula object is the formula object in an array.

Reply:
The feeling I'm going for here would be expressed like this in Scala:

```scala
trait Action
case object NoAction extends Action
case class ActionRequired(formula: Formula) extends Action
```

`NO_ACTION`/`NoAction` is a symbol that nothing needs to be done. Its underlying implementation is just an empty array. It could very well be anything else – `false`, `nil` – but it's convenient for the consumers of the function in which it's used for it to be that empty array.

`action_required_for`/`ActionRequired` is a symbol that something has to be done, a simple wrapper that conveys that further action somehow. It could be a separate class as it would be in a statically typed language but I've chosen to implement it as an array because it's convenient for the consumers of the function in which it's used for it to be that array.

I feel that without this in place, it becomes harder to understand what is expected to be done with the output. It'd be just arrays with no clear _meaning_ of the arrays.